File: ruby/easy/set_mismatch.rb
# @param {Integer[]} nums
# @return {Integer[]}
def find_error_nums(nums)
    res = Array.new
    nums.each do |num|
        n = num.abs - 1
        if nums[n] < 0
            res.push(n+1)
        else
            nums[n] = -nums[n]
        end
    end
    nums.each_with_index do |num, i|
        res.push(i+1) if num > 0
    end
    return res
end

File: javascript/easy/setMismatch.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findErrorNums = function(nums) {
    const res = new Array();
    for(let i = 0; i < nums.length; i++) {
        let n = Math.abs(nums[i]);
        if(nums[n-1] < 0) {
            res.push(n);
        } else {
            nums[n-1] = -nums[n-1];
        }
    }
    
    for(let i = 0; i < nums.length; i++) {
        if(nums[i] > 0) {
            res.push(i+1);
        }
    }
    return res;
};
